test/e2e/failure_domains_test.go

```diff
@@ -0,0 +1,109 @@
+//go:build e2e
+// +build e2e
+
+/*
+Copyright 2023 Nutanix
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	corev1 "k8s.io/api/core/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/test/framework/clusterctl"
+
+	infrav1 "github.com/nutanix-cloud-native/cluster-api-provider-nutanix/api/v1beta1"
+)
+
+const (
+	nutanixFailureDomain1NameEnv = "NUTANIX_FAILURE_DOMAIN_1_NAME"
+	nutanixFailureDomain2NameEnv = "NUTANIX_FAILURE_DOMAIN_2_NAME"
+	nutanixFailureDomain3NameEnv = "NUTANIX_FAILURE_DOMAIN_3_NAME"
+)
+
+// Note: Still has "only-for-validation" label.
+var _ = Describe("Nutanix failure domains", Label("capx-feature-test", "failure-domains", "only-for-validation", "slow", "network"), func() {
```

Comment:
could we document at the top general steps we are taking in this test so that its easier for review the expectations

Reply:
Added a description in the e2e test with the expected verification steps